llvm/lib/Target/AIE/AIEBasePipelinerLoopInfo.cpp

```diff
@@ -211,6 +331,45 @@ AIEBasePipelinerLoopInfo::checkLoopControl(MachineInstr *EndLoop) {
   return nullptr;
 }
 
+MachineBasicBlock *AIEBasePipelinerLoopInfo::getLoopStartBlock() {
+  MachineBasicBlock *LoopStartBlock = nullptr;
+  for (auto *Pred : LoopBlock->predecessors()) {
+    if (Pred == LoopBlock)
+      continue;
+    if (LoopStartBlock)
+      return nullptr;
+    LoopStartBlock = Pred;
+  }
+  return LoopStartBlock;
+}
+
+bool AIEBasePipelinerLoopInfo::shouldUseSchedule(SwingSchedulerDAG &SSD,
+                                                 SMSchedule &SMS) {
+  const int PrologCount = SMS.getMaxStageCount();
+  const int StageCount = PrologCount + 1;
+  LLVM_DEBUG(dbgs() << "StageCount = " << StageCount << "\n");
+  if (StageCount <= 1) {
+    LLVM_DEBUG(dbgs() << "PLI: Rejected schedule (StageCount <= 1)\n");
+    return false;
+  }
```

Comment:
I am a bit surprised that this isn't a standard reason for rejection in MachinePipeliner. StageCount = 1 basically says it's not software-pipelined.

Reply:
Indeed, this was already checked, I just refactored to the base class. Do you think it is dead code?

Reply:
This may be a relic from the time I didn't know too many details of the MachinePipeliner. Perhaps it's a speed hack to skip some otherwise redundant processing. Or, I wanted the rejection message to be included in my "grep PLI" summary.